lib/smack/SmackInstGenerator.cpp

```diff
@@ -847,36 +849,47 @@ void SmackInstGenerator::generateUnModeledCall(llvm::CallInst* ci) {
 void SmackInstGenerator::visitIntrinsicInst(llvm::IntrinsicInst& ii) {
   processInstruction(ii);
 
-  static const auto f16UpCast = [this] (CallInst* ci) {
+//(CallInst -> Void) -> [Flags] -> (CallInst -> Void)
+  static const auto conditionalModel = [this] (std::function<void(CallInst*)> modelGenFunc,
+    std::initializer_list<const cl::opt<bool>*> requiredFlags) {
+    return [this, requiredFlags, modelGenFunc] (CallInst* ci) {
+      bool sat = true;
+      for (auto rf : requiredFlags)
+        if (!*rf)
```

Comment:
Instead of searching for `false` using this for loop, couldn't you use some kind of a built-in function for searching over C++ list?